File: compiler/testData/asJava/ultraLightClasses/delegatesWithAnnotations.java
@java.lang.annotation.Retention(value = java.lang.annotation.RetentionPolicy.RUNTIME)
public abstract @interface SimpleAnn /* SimpleAnn*/ {
  @null()
  public abstract java.lang.String value();//  value()

}

@java.lang.annotation.Retention(value = java.lang.annotation.RetentionPolicy.RUNTIME)
public abstract @interface Ann /* Ann*/ {
  @null()
  public abstract SimpleAnn[] t();//  t()

  @null()
  public abstract java.lang.Class<?> z();//  z()

  @null()
  public abstract java.lang.Class<?>[] e();//  e()

  @null()
  public abstract java.lang.String y();//  y()

  @null()
  public abstract kotlin.DeprecationLevel depr();//  depr()

  public abstract int x();//  x()

}

public abstract interface Base /* Base*/ {
  @Ann(x = 1, y = "134", z = String::class, e = {Int::class, Double::class}, depr = DeprecationLevel.WARNING, t = {@SimpleAnn(value = "243"), @SimpleAnn(value = "4324")})
  public abstract void foo(@Ann(x = 2, y = "324", z = Ann::class, e = {Byte::class, Base::class}, depr = DeprecationLevel.WARNING, t = {@SimpleAnn(value = "687"), @SimpleAnn(value = "78")}) @org.jetbrains.annotations.NotNull() java.lang.String);//  foo(java.lang.String)

}

public final class Derived /* Derived*/ implements Base {
  @Ann(x = 1, y = "134", z = java.lang.String.class, e = {int.class, double.class}, depr = kotlin.DeprecationLevel.WARNING, t = {@SimpleAnn(value="243"), @SimpleAnn(value="4324")})
  public void foo(@Ann(x = 2, y = "324", z = Ann.class, e = {byte.class, Base.class}, depr = kotlin.DeprecationLevel.WARNING, t = {@SimpleAnn(value="687"), @SimpleAnn(value="78")}) @org.jetbrains.annotations.NotNull() java.lang.String);//  foo(java.lang.String)

  @null()
  public  Derived(@org.jetbrains.annotations.NotNull() Base);//  .ctor(Base)

}

File: compiler/testData/asJava/ultraLightClasses/delegatesWithAnnotations.kt
// CHECK_BY_JAVA_FILE
import kotlin.reflect.KClass

annotation class SimpleAnn(val value: String)

annotation class Ann(
    val x: Int,
    val y: String,
    val z: KClass<*>,
    val e: Array<KClass<*>>,
    val depr: DeprecationLevel,
    vararg val t: SimpleAnn
)

interface Base {
    @Ann(1, "134", String::class, arrayOf(Int::class, Double::class), DeprecationLevel.WARNING, SimpleAnn("243"), SimpleAnn("4324"))
    fun foo(
        @Ann(2, "324", Ann::class, arrayOf(Byte::class, Base::class), DeprecationLevel.WARNING, SimpleAnn("687"), SimpleAnn("78")) x: String
    )
}

class Derived(b: Base) : Base by b {

}
